### PR TITLE
US24_Free_an_array_from_heap

### PR DESCRIPTION
**Pull request by NistBox**
Made test for freeing an array on a heap.

### CODE REVIEW

File: Memory/Memory_test/test.cpp [hunk 1]
```diff
@@ -227,24 +227,48 @@ namespace API
...
 		{
-			EXPECT_EQ(array[i], expected_values[i]);
+			EXPECT_EQ(p_array[i], expected_values[i]);
 		}
 
 		memory::Free(p_heap);
 		memory::End();
 	}
+
+	TEST(MemoryAPI, FreeArray)
```
Comment (AmberOrange) [inline Memory/Memory_test/test.cpp]: Quick comment what it does (I think I know, but should maybe have some comment)

File: Memory/Memory_test/test.cpp [hunk 2]
```diff
@@ -227,24 +227,48 @@ namespace API
...
+	{
+		const uint MAIN_HEAP_SIZE = 500;
+		const uint SUB_HEAP_SIZE = 100;
+		const uint OBJ_SIZE = 10;
+		const uint OBJ_COUNT = 5;
+		uint expected_values[OBJ_COUNT];
+
+		memory::Initialize(MAIN_HEAP_SIZE);
+		memory::Heap* p_heap = memory::CreateHeap(SUB_HEAP_SIZE);
```
Comment (AmberOrange) [inline Memory/Memory_test/test.cpp]: Is there a chance this doesn't return a pointer? (Returning nullptr?)

File: Memory/Memory_test/test.cpp [hunk 3]
```diff
@@ -227,24 +227,48 @@ namespace API
...
+		int* p_array = p_heap->AllocateArray<int>(OBJ_COUNT);
```
Comment (AmberOrange) [inline Memory/Memory_test/test.cpp]: Same here

File: Memory/Memory_test/test.cpp (hunk 2 above)
Comment (NistBox) [reply Memory/Memory_test/test.cpp]: There is indeed a chance that CreateHeap can return nullptr. This is tested in _CreateAndFreeSubHeap_, and is expected to work in the other testcases like _FreeArray_.

File: Memory/Memory_test/test.cpp (hunk 3 above)
Comment (NistBox) [reply Memory/Memory_test/test.cpp]: (Copy of the last comment)

There is indeed a chance that CreateHeap can return nullptr. This is tested in CreateAndFreeSubHeap, and is expected to work in the other testcases like AllocateArray and FreeArray.

**Pull request merged**